src/store/actions/entity.actions.js

```diff
@@ -121,6 +125,45 @@ export const generateEntity = (entity, parameters, intl) => (
   return dispatch(releaseSyncLock());
 };
 
+export const importEntity = (entityType, entityId, intl) => (
+  dispatch,
+  getState,
+) => {
+  const isSector = entityType === Entities.sector.key;
+  if (dispatch(preventSync(intl, isSector))) {
+    return Promise.resolve();
+  }
+  const state = getState();
+  const entities = importEntityUtil({
+    entityType,
+    entityId,
+    currentSector: currentSectorSelector(state),
+    data: importDataSelector(state),
+  });
+
+  dispatch({
+    type: UPDATED_ENTITIES,
+    entities,
+  });
+
+  const currentSector = currentSectorSelector(state);
+  const existingSector = sectorSelector(state)[currentSector];
+  const newSectorKeys = Object.keys(entities.sector || {});
```

Comment:
References the entity key here. It would look like `Entitites[Entities.sector.key] || {}`

Reply:
Yea but to make it more general (to make it possible to import not only 'sectors' I would do something like this:
```js
  const currentSector = currentSectorSelector(state);
  const newEntitiyKey = Object.keys(entities[entityType] || {});
  if (!newEntitiyKey.length) {
    dispatch(push('/'));
  } else if (isSector) {
    dispatch(push(`/sector/${newEntitiyKey[0]}`));
  } else if (currentSector) {
    dispatch(
      push(`/sector/${currentSector}/${entityType}/${newEntitiyKey[0]}`),
    );
  } else {
    dispatch(push('/'));
  }
```
With this if a sector is imported, the sector is shown, and if an other entity is imported the sidebar is opened for this entity.

Reply:
An example I could think of the top of my head would be if someone wants to generate there own systems and import them.
But you are right, this is not a very likely use-case, I just like it to implement methods a bit more general to have an easier time using them later on. 😃
Should I commit my proposed change or just the use of `Entitites`?